### DISCUSSION TITLE
How to create service for autostarting postgres after installation of main package using pg_ctl and this service should autostop after uninstallation of main package

### DISCUSSION THREAD

**Discussion by vaa2000** [wixtoolset/issues#8620] [Questions]
Following is the code that I have included using CustomActions in wix toolset - as passing arguments with dynamic folder location was not working with ServiceInstall and ServiceControl tags in wix toolset can you please suggest why the following code is not working to autostart and autostop services-

    <CustomAction Id="RegisterPgCtlService" Directory="directory2" Execute="deferred" Return="ignore" Impersonate="no" ExeCommand="[directory2]pg_ctl.exe register -N &quot;iVisionmax Historical Database&quot; -D &quot;[INSTALLLOCATION]db\LThistdb\data&quot; -W" />
    <CustomAction Id="StartPgCtlService" Directory="directory2" Execute="deferred" Return="ignore" Impersonate="no" ExeCommand="[directory2]pg_ctl.exe start -N &quot;iVisionmax Historical Database&quot; -D &quot;[INSTALLLOCATION]db\LThistdb\data&quot; -W" />
    <CustomAction Id="StopPgCtlService" Directory="directory2" Execute="deferred" Return="ignore" Impersonate="no" ExeCommand="[directory2]pg_ctl.exe stop -N &quot;iVisionmax Historical Database&quot; -D &quot;[INSTALLLOCATION]db\LThistdb\data&quot; -W" />
    <CustomAction Id="UnregisterPgCtlService" Directory="directory2" Execute="deferred" Return="ignore" Impersonate="no" ExeCommand="[directory2]pg_ctl.exe unregister -N &quot;iVisionmax Historical Database&quot;" />

    <InstallExecuteSequence>
      <Custom Action="RegisterPgCtlService" After="InstallFiles">NOT REMOVE</Custom>
      <Custom Action="StartPgCtlService" After="RegisterPgCtlService">NOT REMOVE</Custom>
      <Custom Action="StopPgCtlService" After="RemoveFiles">REMOVE="ALL"</Custom>
      <Custom Action="UnregisterPgCtlService" Before="StopPgCtlService">REMOVE="ALL"</Custom>
    </InstallExecuteSequence>

**Comment by bevanweiss** [comment]
Your use of CustomAction isn't right.
https://wixtoolset.org/docs/v3/xsd/wix/customaction/
ExeCommand does not actually execute anything... it's the command to provide TO the executable (not FOR execution)
You will probably want to look into the Quiet Execution custom actions
https://wixtoolset.org/docs/v3/customactions/qtexec/

**Reply by bevanweiss** [reply]
ServiceInstall should have worked though
https://wixtoolset.org/docs/v3/xsd/wix/serviceinstall/
https://learn.microsoft.com/en-au/windows/win32/msi/serviceinstall-table?redirectedfrom=MSDN

The Arguments column is of Formatted type.
You would probably want to use Orca to confirm that WiX hasn't done anything it shouldn't in populating the table.
i.e. that the formatted value you expected to end up in the ServiceInstall table's column was actually as you had provided.
